modules/local/pull_references.nf

```diff
@@ -0,0 +1,38 @@
+// Import generic module functions
+include { getSoftwareName } from './functions'
+
+
+process BLAST_BLASTDBCMD{
+    tag "$sample_name - Segment:$segment - Ref Accession ID:$id"
+    label 'process_medium'
+    publishDir "${params.outdir}/reference_sequences/$sample_name",
+        pattern: "*.fasta",
+        mode: params.publish_dir_mode
+
+    conda (params.enable_conda ? 'bioconda::blast=2.10.1' : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container 'https://depot.galaxyproject.org/singularity/blast:2.10.1--pl526he19e7b1_3'
+    } else {
+        container 'quay.io/biocontainers/blast:2.10.1--pl526he19e7b1_3'
+    }
+
+    input:
+    tuple val(sample_name), val(segment), val(id), path(reads)
+    path (db)
+
+    output:
+    tuple val(sample_name), val(segment), val(id), path('*.fasta'), path(reads), emit: fasta
+    path '*.version.txt'            , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = "${sample_name}"
+    """
+    DB=`find -L ./ -name "*.ndb" | sed 's/.ndb//'`
+    blastdbcmd \\
```

Comment:
Use [seqtk](https://github.com/lh3/seqtk/) on the influenza fasta instead.

```bash
seqtk subseq influenza.fna seqs-to-get.txt > ref.fasta
# or if single seq
echo "seqid" | seqtk subseq influenza.fna - > ref.fasta
```

Reply:
Updated